src/ImageSharp/Formats/AnimationUtilities.cs

```diff
@@ -0,0 +1,190 @@
+    /// <param name="resultFrame">The resultant output.</param>
+    /// <param name="replacement">The value to use when replacing duplicate pixels.</param>
+    /// <param name="clampingMode">The clamping bound to apply when calculating difference bounds.</param>
+    /// <returns>The <see cref="ValueTuple{Boolean, Rectangle}"/> representing the operation result.</returns>
+    public static (bool Difference, Rectangle Bounds) DeDuplicatePixels<TPixel>(
+        Configuration configuration,
+        ImageFrame<TPixel>? previousFrame,
+        ImageFrame<TPixel> currentFrame,
+        ImageFrame<TPixel> resultFrame,
+        Vector4 replacement,
+        ClampingMode clampingMode = ClampingMode.None)
+        where TPixel : unmanaged, IPixel<TPixel>
+    {
+        // TODO: This would be faster (but more complicated to find diff bounds) if we operated on Rgba32.
+        // If someone wants to do that, they have my unlimited thanks.
+        MemoryAllocator memoryAllocator = configuration.MemoryAllocator;
+        IMemoryOwner<Vector4> buffers = memoryAllocator.Allocate<Vector4>(currentFrame.Width * 3, AllocationOptions.Clean);
+        Span<Vector4> previous = buffers.GetSpan()[..currentFrame.Width];
+        Span<Vector4> current = buffers.GetSpan().Slice(currentFrame.Width, currentFrame.Width);
+        Span<Vector4> result = buffers.GetSpan()[(currentFrame.Width * 2)..];
+
+        int top = int.MinValue;
+        int bottom = int.MaxValue;
+        int left = int.MaxValue;
+        int right = int.MinValue;
+
+        bool hasDiff = false;
+        for (int y = 0; y < currentFrame.Height; y++)
+        {
+            if (previousFrame != null)
+            {
+                PixelOperations<TPixel>.Instance.ToVector4(configuration, previousFrame.DangerousGetPixelRowMemory(y).Span, previous, PixelConversionModifiers.Scale);
+            }
+
+            PixelOperations<TPixel>.Instance.ToVector4(configuration, currentFrame.DangerousGetPixelRowMemory(y).Span, current, PixelConversionModifiers.Scale);
+
+            ref Vector256<float> previousBase = ref Unsafe.As<Vector4, Vector256<float>>(ref MemoryMarshal.GetReference(previous));
+            ref Vector256<float> currentBase = ref Unsafe.As<Vector4, Vector256<float>>(ref MemoryMarshal.GetReference(current));
+            ref Vector256<float> resultBase = ref Unsafe.As<Vector4, Vector256<float>>(ref MemoryMarshal.GetReference(result));
+
+            Vector256<float> replacement256 = Vector256.Create(replacement.X, replacement.Y, replacement.Z, replacement.W, replacement.X, replacement.Y, replacement.Z, replacement.W);
+
+            int size = Unsafe.SizeOf<Vector4>();
+
+            bool hasRowDiff = false;
+            int i = 0;
+            uint x = 0;
+            int length = current.Length;
+            int remaining = current.Length;
+            if (Avx2.IsSupported && remaining >= 2)
+            {
+                while (remaining >= 2)
+                {
```

Comment:
So the condition `remaining >= 2` is check twice. Thus either make a `do-while` loop or keep the `while` but as
```suggestion
                while (Avx2.IsSupported && remaining >= 2)
                {
```
. The JIT will constant eleminate `Avx2.IsSupported`.
Personally I'd go with the `do-while` as it reads more natural then.